plugin-api/src/main/java/org/hyperledger/besu/plugin/data/ProcessableBlockHeader.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.plugin.data;
+
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.datatypes.Quantity;
+import org.hyperledger.besu.plugin.Unstable;
+
+import java.util.Optional;
+
+import org.apache.tuweni.bytes.Bytes32;
+
+/**
+ * The minimum set of data for a BlockHeader, as defined in the <a href=
+ * "https://ethereum.github.io/yellowpaper/paper.pdf">Ethereum Yellow Paper</a>.
+ */
```

Comment:
nit: we may need a better description for this interface. It is not exactly the minimum set of data but a even smaller part of the header used during block construction.